khl/rule.py

```diff
@@ -0,0 +1,46 @@
+from typing import Callable, Coroutine
+from abc import ABC, abstractmethod
+from khl import Msg
+
+
+class Rule(ABC):
+    @abstractmethod
+    def at_me(*args, **kwargs) -> (Callable[..., Coroutine]):
+        raise NotImplementedError
+
+    @abstractmethod
+    def at_all(*args, **kwargs) -> (Callable[..., Coroutine]):
+        raise NotImplementedError
+
+    @abstractmethod
+    def on_channel(self, name: str, *args, **kwargs) -> (Callable[..., Coroutine]):
+        raise NotImplementedError
+
+
+class MessageRule(Rule):
+    """
+    can used in all message
+
+    @bot.command(name='hello', rule=MessageRule.at_me())
+    async def examples(msg: TextMsg, *args):
+        await msg.reply(f'hello')
+    """
+
+    def at_me(*args, **kwargs) -> (Callable[..., Coroutine]):
```

Comment:
first arg of method should be `self`

i think you mean
```python
@staticmethod
def at_me(*args, **kwargs):
```

Reply:
besides, these check functions is simple enough to be normal sync rather async, a bit misleading and unreadable

Reply:
if i have a function to wait,async function it works
or there are other solutions? i not familiar with py

Reply:
> if i have a function to wait,async function it works
> or there are other solutions? i not familiar with py

depends on the situation:

> assume the function to wait is `func()`

1. wait in another sync func: 
    - func() is sync: call it directly
    - func() is async: `asyncio.get_event_loop().run_until_complete(func())`
2. wait in another async func:
    - func() is sync: call it directly
    - func() is async: `await func()`